File: src/api/LMSEntities/DataTransferObjects/LibraryAssetForCreationDto.cs
using System;
using System.Collections.Generic;

namespace LMSEntities.DataTransferObjects
{
    public class LibraryAssetForCreationDto
    {
        public string DeweyIndex { get; set; }

        public LibraryAssetTypeDto AssetType { get; set; }

        public ICollection<LibraryAssetAuthorDto> AssetAuthors { get; set; }

        public ICollection<LibraryAssetCategoryDto> AssetCategories { get; set; }

        public int NumberOfCopies { get; set; }

        public string Description { get; set; }

        public string ISBN { get; set; }

        public string Title { get; set; }

        public int Year { get; set; }
    }
}

File: src/spa/src/dto/models/library-asset-for-creation-dto.ts
/* tslint:disable */
/* eslint-disable */
import { LibraryAssetAuthorDto } from './library-asset-author-dto';
import { LibraryAssetCategoryDto } from './library-asset-category-dto';
import { LibraryAssetTypeDto } from './library-asset-type-dto';
export interface LibraryAssetForCreationDto {
  assetAuthors: Array<LibraryAssetAuthorDto>;
  assetCategories: Array<LibraryAssetCategoryDto>;
  assetType?: LibraryAssetTypeDto;
  description?: null | string;
  deweyIndex?: null | string;
  isbn?: null | string;
  numberOfCopies?: number;
  title: string;
  year: number;
}
